plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbTypeMapping.java

```diff
@@ -876,6 +841,99 @@ public void testTimestampCoercion()
                 .execute(getQueryRunner(), trinoCreateAndInsert("test_timestamp_coercion"));
     }
 
+    @Test
+    public void testDatetime()
+    {
+        testDatetime(UTC);
+        testDatetime(jvmZone);
+        testDatetime(vilnius);
+        testDatetime(kathmandu);
+        testDatetime(TestingSession.DEFAULT_TIME_ZONE_KEY.getZoneId());
+    }
+
+    private void testDatetime(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(getTimeZoneKey(sessionZone.getId()))
+                .build();
+        timestampRoundTrips("datetime")
+                // max value 9999-12-31 23:59:59
+                .addRoundTrip("datetime(3)", "TIMESTAMP '9999-12-31 23:59:59.999'", createTimestampType(3), "TIMESTAMP '9999-12-31 23:59:59.999'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '9999-12-31 23:59:59.999999'", createTimestampType(6), "TIMESTAMP '9999-12-31 23:59:59.999999'")
+                // min value 1000-01-01 00:00:00
+                .addRoundTrip("datetime(3)", "TIMESTAMP '1000-01-01 00:00:00.000'", createTimestampType(3), "TIMESTAMP '1000-01-01 00:00:00.000'")
```

Comment:
This value isn't the min value in MariaDB as far as I confirmed:
```
create table tpch.test(a datetime);
insert into tpch.test values (TIMESTAMP '0999-12-31 00:00:00.000');
select * from tpch.test; -- 0999-12-31 00:00:00.000
```

Reply:
Ah it is supporting min value of `0001-01-01 00:00:00`,  i followed this doc https://mariadb.com/kb/en/datetime/#supported-values.

Reply:
updated